### PR TITLE
151 additional

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Fixes for generating equipment catalogue function in dev environments.

### CODE REVIEW

File: src/equipment/equipmentUtils.ts [hunk 1]
```diff
@@ -165,13 +165,17 @@ export async function attachEquipmentToExecutionRequirement(
 
 export async function getRandomEquipmentData(
   operatorId: number,
-  excludeVehicleIds: (string | undefined)[] = []
+  excludeVehicleIds: (string | undefined)[] = [],
+  type?: string
```
Comment (Divinare) [inline src/equipment/equipmentUtils.ts:169]: Would be readable when this method is called in format:
```
export async function getRandomEquipmentData({
  operatorId,
  excludeVehicleIds,
  type
} : {
  operatorId: number,
  excludeVehicleIds: (string | undefined)[] = [],
  type?: string
})
```

File: src/equipment/equipmentUtils.ts [hunk 2]
```diff
@@ -165,13 +165,17 @@ export async function attachEquipmentToExecutionRequirement(
...
   let equipmentData: DeepPartial<Equipment> | undefined
   let i = 0
 
-  while (i < 10 && (!equipmentData || excludeVehicleIds.includes(equipmentData?.vehicleId || ''))) {
-    equipmentData = (await Equipment.getRandomEquipmentOfOperator(operatorId)) || undefined
+  while (
+    i < 100 &&
+    (!equipmentData || excludeVehicleIds.includes(equipmentData?.vehicleId || ''))
+  ) {
+    equipmentData = (await Equipment.getRandomEquipmentOfOperator(operatorId, type)) || undefined
```
Comment (Divinare) [inline src/equipment/equipmentUtils.ts:178]: I think we should avoid while loop here. But could this be made without having to exclude vehicle ids?

**Comment by Divinare** [comment]
PR ok (left 1 comment). Didn't look into generate equipment func so much, as long as it works its ok (btw a test that validates it would be awesome)

File: src/equipment/equipmentUtils.ts (hunk 1 above)
Comment (danieldunderfelt) [reply src/equipment/equipmentUtils.ts:169]: I don't think using objects here is wrong, but separate arguments also work just fine. If it ain't broke there's no need to fix it.

File: src/equipment/equipmentUtils.ts (hunk 2 above)
Comment (danieldunderfelt) [reply src/equipment/equipmentUtils.ts:178]: The challenge is to get a random vehicle that does not already exist in the catalogue. So it does it until a valid vehicle is returned. There's nothing wrong with while loops, and they work well in these cases. I don't see how it is possible to exclude vehicle ID's without having code that does it. Maybe I am misunderstanding you?

This is also just a dev tool so it doesn't have to be the most polished implementation.

**Comment by danieldunderfelt** [comment]
> PR ok (left 1 comment). Didn't look into generate equipment func so much, as long as it works its ok (btw a test that validates it would be awesome)

Heh yeah it's a bit complicated. One could add a check at the end that all vehicle quotas add up to a 100%, but it's not hugely important imo.

**Pull request merged**